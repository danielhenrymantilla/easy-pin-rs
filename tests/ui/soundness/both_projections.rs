#![allow(unused)]

use ::easy_pin::{
    easy_pin,
    PinSensitive,
};
use ::core::ptr::NonNull;

#[easy_pin]
pub
struct SelfReferential {
    #[transitively_pinned]
    #[unpinned]
    string: PinSensitive<String>,

    #[unpinned]
    at_string: NonNull<String>,
}

fn main ()
{}
