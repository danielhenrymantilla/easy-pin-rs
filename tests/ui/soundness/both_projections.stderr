error: #[unpinned] / #[transitively_pinned] must be specified exactly once per field
  --> $DIR/both_projections.rs:13:7
   |
13 |     #[unpinned]
   |       ^^^^^^^^
